com.unity.multiplayer.mlapi/Tests/Runtime/NetworkObject/NetworkObjectDestroyTests.cs

```diff
@@ -0,0 +1,245 @@
+
+            Assert.That(MultiInstanceHelpers.Create(k_ClientInstanceCount, out m_ServerNetworkManager, out m_ClientNetworkManagers));
+            Assert.That(m_ServerNetworkManager, Is.Not.Null);
+            Assert.That(m_ClientNetworkManagers, Is.Not.Null);
+            Assert.That(m_ClientNetworkManagers.Length, Is.EqualTo(k_ClientInstanceCount));
+
+            m_PlayerPrefab = new GameObject("PlayerPrefabPrototype");
+            MultiInstanceHelpers.MakeNetworkedObjectTestPrefab(m_PlayerPrefab.AddComponent<NetworkObject>());
+
+            m_DummyPrefab = new GameObject("DummyPrefabPrototype");
+            MultiInstanceHelpers.MakeNetworkedObjectTestPrefab(m_DummyPrefab.AddComponent<NetworkObject>());
+
+            m_ServerNetworkManager.NetworkConfig.PlayerPrefab = m_PlayerPrefab;
+            m_ServerNetworkManager.NetworkConfig.NetworkPrefabs.Add(new NetworkPrefab { Prefab = m_DummyPrefab });
+            foreach (var clientNetworkManager in m_ClientNetworkManagers)
+            {
+                clientNetworkManager.NetworkConfig.PlayerPrefab = m_PlayerPrefab;
+                clientNetworkManager.NetworkConfig.NetworkPrefabs.Add(new NetworkPrefab { Prefab = m_DummyPrefab });
+            }
+
+            Assert.That(MultiInstanceHelpers.Start(/* isHost = */ true, m_ServerNetworkManager, m_ClientNetworkManagers));
+            yield return MultiInstanceHelpers.Run(MultiInstanceHelpers.WaitForClientsConnected(m_ClientNetworkManagers));
+            yield return MultiInstanceHelpers.Run(MultiInstanceHelpers.WaitForClientConnectedToServer(m_ServerNetworkManager));
+        }
+
+        [TearDown]
+        public void Teardown()
+        {
+            MultiInstanceHelpers.Destroy();
+
+            if (m_PlayerPrefab != null)
+            {
+                Object.DestroyImmediate(m_PlayerPrefab);
+            }
+
+            if (m_DummyGameObject != null)
+            {
+                Object.DestroyImmediate(m_DummyGameObject);
+            }
+
+            if (m_DummyPrefab != null)
+            {
+                Object.DestroyImmediate(m_DummyPrefab);
+            }
+        }
+
+        [UnityTest]
+        public IEnumerator ShouldDestroyPlayerObjectOnDisconnect()
+        {
+            var c0PlayerNetworkObject = m_ServerNetworkManager.ConnectedClientsList[0].PlayerObject;
+            var c0PlayerGameObject = c0PlayerNetworkObject.gameObject;
+            var c0PlayerObjectId = c0PlayerNetworkObject.NetworkObjectId;
+            var c0PlayerClientId = m_ServerNetworkManager.ConnectedClientsList[0].ClientId;
+
+            Assert.That(m_ServerNetworkManager.SpawnManager.SpawnedObjects.ContainsKey(c0PlayerObjectId));
+            foreach (var clientNetworkManager in m_ClientNetworkManagers)
+            {
+                Assert.That(clientNetworkManager.SpawnManager.SpawnedObjects.ContainsKey(c0PlayerObjectId));
+            }
+
+
+            m_ServerNetworkManager.DisconnectClient(c0PlayerClientId);
```

Comment:
apparently, this call to `SIPTransport` is not causing PlayerObject to delete and client disconnect callback to be fired.
there should be some issues sitting somewhere behind `SIPTransport` @TwoTenPvP

Reply:
Seems like a SIP bug. Disconnects were not thouroughly tested